src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/tab.cljs

```diff
@@ -51,88 +52,105 @@
                      (swap! state assoc :server-error (get-parsed-response))))}))})
 
 
-(defn- render-overlays [state]
-  [:div {}
-   (when (:deleting-attrs? @state)
-     [comps/Blocker {:banner "Deleting Attributes..."}])
-   (when (:updating-attrs? @state)
-     [comps/Blocker {:banner "Updating Attributes..."}])
-   (when (contains? @state :locking?)
-     [comps/Blocker {:banner (if (:locking? @state) "Unlocking..." "Locking...")}])])
+(defn- save-attributes [{:keys [old-attributes new-attributes state workspace-id request-refresh]}]
+  (let [to-delete (->> (clojure.set/difference (set (keys old-attributes)) (set (keys new-attributes)))
+                       (map (fn [key] {:op "RemoveAttribute" :attributeName key})))
+        to-update (->> new-attributes
+                       (filter (fn [[k v]] (not= v (get old-attributes k))))
+                       (map (fn [[k v]] {:op "AddUpdateAttribute" :attributeName k :addUpdateAttribute v})))]
+    (swap! state assoc :updating-attrs? true)
+    (endpoints/call-ajax-orch
+      {:endpoint (endpoints/update-workspace-attrs workspace-id)
+       :payload (concat to-delete to-update)
+       :headers utils/content-type=json
+       :on-done (fn [{:keys [success? get-parsed-response]}]
+                  (swap! state dissoc :updating-attrs? :editing?)
+                  (if success?
+                    (request-refresh)
+                    (modal/push-error-response (get-parsed-response))))})))
```

Comment:
for a future story: I'd still like to see the diffing and calculation of the RemoveAttribute/AddUpdateAttribute operations handled in orchestration; it shouldn't be necessary here.